modules/marshaller-common/src/main/java/org/apache/ignite/internal/marshaller/MarshallerColumn.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.marshaller;
+
+import java.util.function.Supplier;
+import org.apache.ignite.internal.tostring.IgniteToStringExclude;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Marshaller column.
+ */
+public class MarshallerColumn {
```

Comment:
This is already a 3-rd class for the same purpose, and same content.
Column, ClientColumn, and MarshallerColumn all of them are internal classes representing a column descriptor.

Reply:
Those 3 classes have different responsibilities. `ClientColumn` won't have all of the information that server-side `Column` has, so I don't think we can use a single class.

We can change `MarshallerColumn` to be an interface though, and implement that interface in `Column` and `ClientColumn`. We can do that as part of code reuse in https://issues.apache.org/jira/browse/IGNITE-16088